modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/ObjectRelationshipInheritanceCheckbox.tsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayCheckbox} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayPopover from '@clayui/popover';
+import React, {useState} from 'react';
+
+import './ObjectRelationshipInheritanceCheckbox.scss';
+
+interface ObjectRelationshipInheritanceCheckbox {
+	onSubmit: (values?: Partial<ObjectRelationship>) => Promise<void>;
+	setValues: (values: Partial<ObjectRelationship>) => void;
+	values: Partial<ObjectRelationship>;
+}
+
+export function ObjectRelationshipInheritanceCheckbox({
+	onSubmit,
+	setValues,
+	values,
+}: ObjectRelationshipInheritanceCheckbox) {
+	const [showPopover, setShowPopover] = useState(false);
+
+	return (
+		<>
+			<div className="form-group lfr__object-relationship-inheritance-container">
+				<ClayCheckbox
+					checked={!!values.edge}
+					label={Liferay.Language.get(
+						'enable-inheritance-including-shared-permissions-and-grouped-apis'
+					)}
+					onChange={({target}) => {
+						if (target.checked) {
+							setValues({
+								...values,
+								edge: true,
+							});
+						}
+						else {
+							const parentWindow = Liferay.Util.getOpener();
+
+							parentWindow.Liferay.fire(
+								'openModalDisableInheritance',
+								{
+									handleDisable: async () => {
+										setValues({
```

Comment:
There many instances of calling onSubmit after setting values, but most are conditioned by autosave, which is different. Please check other calls to onSubmit and see if my above comment makes sense. Thanks!

Reply:
Hey, I was able to do some research and noticed that we have 20 occurrences of the onSubmit function in object-web and by default they return vold. I think it might be nice to continue following this same pattern in this case. 

What I thought could be done to solve this case of the error you mentioned is just remove the setValue call.

In case the API returns success, the submit function is responsible for saving and reloading the page, we will reconstruct the states based on the new API returns.

In the case of an error, we will display the error message and no state change will have been made.

What do you say?

Reply:
Perhaps making the submit functions return is the best thing to do. But I think we should think more about this and do it in a coordinated way that covers all cases.

Reply:
yeah, because the page is reloaded, no setValues is necessary.